### PR TITLE
Mongo setup

### LINKED ISSUE #2: Setup Bazy danych
Zrobić wstępny setup bazy danych i nawiązać połączenie baza - backend

- [ ] https://github.com/AtmosFears/AtmosFears-BE/issues/10
- [ ] https://github.com/AtmosFears/AtmosFears-BE/issues/8

### LINKED ISSUE #8: Connect with database
For this task, you should create a class that will manage database operations (for now, get data and insert data)

Requirements:
https://github.com/AtmosFears/AtmosFears-BE/issues/10

### LINKED ISSUE #10: MongoDB setup
1. Setup mongoDB
2. Describe in README how to connect to DB

### PR DESCRIPTION
**Pull request by pgorgolew**
image

### CODE REVIEW

File: src/main/java/atmosfears/AtmosFearsBE/controller/HelloController.java [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 
 @RestController
 public class HelloController {
 
+    @Autowired
+    private AirParticulatesService airParticulatesService;
     @GetMapping("/")
     public String index() {
-        return "Greetings from Spring Boot!";
+        return "Greetings from Spring Boot! <br> Repository findAll result: " + airParticulatesService.findAll();
```
Comment (MatiPl01) [inline src/main/java/atmosfears/AtmosFearsBE/controller/HelloController.java]: You don't have to use the `@Autowired` decorator. It's even not recommended and the better approach is to use constructor injection instead.

You can replace selected lines of code with the following code:

```ts
private final AirParticulatesService airParticulatesService;

private HelloController(AirParticulatesService airParticulatesService) {
    this.airParticulatesService = airParticulatesService;
}
```

File: src/main/java/atmosfears/AtmosFearsBE/controller/HelloController.java [hunk 2]
```diff
@@ -1,14 +1,18 @@
 package atmosfears.AtmosFearsBE.controller;
 
+import atmosfears.AtmosFearsBE.service.AirParticulatesService;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (MatiPl01) [inline src/main/java/atmosfears/AtmosFearsBE/controller/HelloController.java]:
```suggestion
```
Remove this import after changing to the `constructor` injection.

File: src/main/java/atmosfears/AtmosFearsBE/service/AirParticulatesService.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Service
+public class AirParticulatesService {
+
+    @Autowired
```
Comment (MatiPl01) [inline src/main/java/atmosfears/AtmosFearsBE/service/AirParticulatesService.java]:
```suggestion
```
`@Autowired` is not needed in here as well. The `@AllArgsConstructor` from `lombok` creates all constructors and one of them is a constructor that takes the `AirParticulatesRepository` as a parameter. The `constructor` injection will be performed automatically, thanks to the constructor created by `@AllArgsConstructor` so we don't need to inject this repository manually.

File: src/main/java/atmosfears/AtmosFearsBE/service/AirParticulatesService.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package atmosfears.AtmosFearsBE.service;
+
+import atmosfears.AtmosFearsBE.database.AirParticulatesRepository;
+import atmosfears.AtmosFearsBE.model.AirParticulates;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (MatiPl01) [inline src/main/java/atmosfears/AtmosFearsBE/service/AirParticulatesService.java]:
```suggestion
```

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1 +1,2 @@
-
+spring.data.mongodb.uri=mongodb+srv://${db.username}:${db.password}@atmosfearscluser.ommn72o.mongodb.net
```
Comment (MatiPl01) [inline src/main/resources/application.properties:1]: Will you create users in the MongoDB on Atlas and give us access to the same database in cloud?

**Review by MatiPl01 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 5 above)
Comment (pgorgolew) [reply src/main/resources/application.properties:1]: Right now there is only one user that can be used by every member of organization. We should discuss if we want to create user per person or leave it like that

**Comment by pgorgolew** [comment]
image
image

I inserted one example by new method `insertEmpty` .  It works both with spring and MongoDB Cluster. Uploading screens to show it is working.

**Review by MatiPl01 — APPROVED** [review]
LGTM

**Review by kacperklusek — APPROVED** [review]

**Pull request merged**